packages/hash/api/src/graphql/resolvers/user/updateUser.ts

```diff
@@ -1,83 +1,98 @@
+    throw new UserInputError("Shortname must be at least 4 characters long.");
+  }
+  if (shortname.length > shortnameMaximumLength) {
+    throw new UserInputError("Shortname cannot be longer than 24 characters");
+  }
+};
+
 export const updateUser: ResolverFn<
   Promise<UnresolvedGQLUser>,
   {},
   LoggedInGraphQLContext,
   MutationUpdateUserArgs
-> = async (_, { userEntityId, properties }, { dataSources, user }) =>
-  dataSources.db.transaction(async (client) => {
-    const { graphApi } = dataSources;
-    /** @todo: allow HASH admins to bypass this */
+> = async (
+  _,
+  { userEntityId, properties },
+  { dataSources: { graphApi }, user },
+) => {
+  /** @todo: run this mutation in a transaction */
 
-    if (userEntityId !== user.entityId) {
-      throw new ForbiddenError("You can only update your own user properties");
-    }
+  /** @todo: allow HASH admins to bypass this */
+  if (userEntityId !== user.entityId) {
+    throw new ForbiddenError("You can only update your own user properties");
+  }
 
-    const { shortname, preferredName, usingHow } = properties;
+  let updatedUser = user;
 
-    if (!shortname && !preferredName && !usingHow) {
-      throw new ApolloError(
-        "An updated 'shortname', 'preferredName' or 'usingHow' value  must be provided to update a user",
-        "NO_OP",
-      );
-    }
+  const { shortname, preferredName } = properties;
 
-    if (shortname) {
-      if (user.getShortname() === shortname && !preferredName) {
-        throw new ApolloError(
-          `User with entityId '${user.entityId}' already has the shortname '${shortname}'`,
-          "NO_OP",
-        );
-      }
+  if (shortname && user.getShortname() !== shortname) {
+    await validateShortname(graphApi, shortname);
 
-      await Account.validateShortname(client, shortname);
+    updatedUser = await updatedUser.updateShortname(graphApi, {
+      updatedByAccountId: user.accountId,
+      updatedShortname: shortname,
+    });
+  }
 
-      await user.updateShortname(graphApi, {
-        updatedByAccountId: user.accountId,
-        updatedShortname: shortname,
-      });
-    }
-
-    if (preferredName) {
-      if (user.getPreferredName() === preferredName) {
-        throw new ApolloError(
-          `User with entityId '${user.entityId}' already has the preferredName '${preferredName}'`,
-          "NO_OP",
-        );
-      }
-
-      if (!User.preferredNameIsValid(preferredName)) {
-        throw new ApolloError(
-          `The preferredName '${preferredName}' is invalid`,
-          "PREFERRED_NAME_INVALID",
-        );
-      }
-
-      await user.updatePreferredName(graphApi, {
-        updatedByAccountId: user.accountId,
-        updatedPreferredName: preferredName,
-      });
+  if (preferredName && user.getPreferredName() !== preferredName) {
+    if (UserModel.preferredNameIsInvalid(preferredName)) {
+      throw new ApolloError(
+        `The preferredName '${preferredName}' is invalid`,
```

Comment:
This doesn't seem to actually trigger for me when trying to update a `preferredName` to `""`. Also we have the same check for invalid names in `updatedUser.updatePreferredName`. Is this duplication done to throw an apollo error isntead?

Reply:
Hmm good point, this is a JS thing: `""` evaluates to falsy this method assumes that you're not trying to set the preferred name. Probably best to check for `undefined`, and throw if `""` is provided. Have done so in https://github.com/hashintel/hash/pull/950/commits/f427cdd85a7c404fe5a9e74886cfb072987853ba, let me know what you think.

Reply:
I think this is fine :+1:

Reply:
It's a little bit awkward because the `updateUser` method is designed to be used for updating some or multiple fields of a user, depending on the context. I still think this is the way to go though.